### ISSUE TITLE
UiAutocomplete 

### ISSUE THREAD

**Issue by EmilMoe** [JosephusPaye/Keen-UI#15]
I'm not sure where you define the value prop? But it's returning the text value. I can't see how to return the value property which contains my id?

Ideally I would like to:
- :text.sync=""
- - to sync the text 
- :value.sync=""
- - to sync the value
- :object.sync=""
- - to sync the whole object

**Comment by JosephusPaye** [comment]
The behavior you are describing (where the text and value are synced separately) was removed when I rewrote the autocomplete in v0.8.6. 

Now it only only auto-completes the user's input, therefore there is only one prop to watch, the value.

What you want is more like a select component, which I plan on adding soon.

**Comment by EmilMoe** [comment]
I want a select too yes. But I use the auto complete to search for collections in the database like users. I want the users name to be displayed, but I want to submit the users ID.

**Comment by JosephusPaye** [comment]
In that case you could use the text (value) to find the ID if you have it in a suggestions array. This is how it was done in [v0.8.5](https://github.com/JosephusPaye/keen-ui/blob/87287f1cd35e19ede0739e6f9883cd9e799c753f/src/UiAutocomplete.vue#L246):

```js
function findValue(text) {
    // Just return the text if suggestions is an array of strings
    if (this.suggestions[0] && typeof this.suggestions[0] === 'string') {
        return text;
    }
    text = text.toLowerCase();
    for (let i = 0; i < this.suggestions.length; i++) {
        if (this.suggestions[i].text.toLowerCase() === text) {
            return this.suggestions[i].value;
        }
    }
    return null;
}
```

You can then run this function on submit to find the user ID. The new select component will have this functionality by default.

**Comment by EmilMoe** [comment]
I'm not sure if we differentiate a select and an auto complete the same way?

For me a select has a preset range of values and don't have a search option
The auto complete has a dynamic set of values and they are found by searching

**Comment by JosephusPaye** [comment]
A select can also be dynamic and have search. Like, for example, [Select2](https://select2.github.io/examples.html). 

But a select doesn't allow the user to select a value that's not in the options, while an autocomplete allows for entering any text, even if it's not in the suggestions.

**Comment by EmilMoe** [comment]
That's true.

Do you have an idea when you have it finished?

**Comment by JosephusPaye** [comment]
Not sure, but I've already started work on it and it might be done by Friday.

**Comment by EmilMoe** [comment]
Perfect :-)

**Comment by athikrishnan** [comment]
Not to rush but i could really use a select component in the coming week. going to use keen-ui on a product.

**Comment by JosephusPaye** [comment]
A [Select](http://josephuspaye.github.io/Keen-UI/#/ui-select-docs) and [Slider](http://josephuspaye.github.io/Keen-UI/#/ui-slider-docs) component just landed in [v0.8.8](https://github.com/JosephusPaye/Keen-UI/releases/tag/v0.8.8). Check it out.
